stdlib/public/core/Sequence.swift

```diff
@@ -428,29 +427,20 @@ extension Sequence {
       return result
     }
 
-    var hitEnd = false
     var iterator = self.makeIterator()
-    while true {
-      guard let element = iterator.next() else {
-        hitEnd = true
-        break
-      }
+    while let element = iterator.next() {
       if try isSeparator(element) {
-        if !appendSubsequence() {
-          continue
-        }
-        if result.count == maxSplits {
-          break
-        }
-      } else {
-        subSequence.append(element)
+        appendSubsequence()
       }
-    }
-    if !hitEnd {
-      while let element = iterator.next() {
-        subSequence.append(element)
+      if result.count == maxSplits {
+        break
       }
     }
+
+    while let element = iterator.next() {
```

Comment:
This code relies on the fact that `iterator.next()` would keep returning `nil` after it has done it once, which is not guaranteed as of now. Having said that, [the proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0052-iterator-post-nil-guarantee.md) to introduce this kind of contract is under review until today.